lib/std/stackframes.nim

```diff
@@ -0,0 +1,30 @@
+const NimStackTrace = compileOption("stacktrace")
+
+template procName*(): string =
+  ## returns current C/C++ function name
+  when defined(c) or defined(cpp):
+    var name {.inject.}: cstring
+    {.emit: "`name` = __func__;".}
+    $name
+
+template getPFrame*(): PFrame =
+  ## avoids a function call (unlike `getFrame()`)
+  block:
+    when NimStackTrace:
+      var framePtr {.inject.}: PFrame
```

Comment:
1. Is this function call really a problem?
2. Shouldn't this be ``var framePtr {.importc: "FR_", nodecl.}: PFrame; framePtr``?

Reply:
TLDR:
1: probably
2: that doesn't work

1. considering it can get called in high frequency, and there's no hard guarantee C compiler will inline despite hints, and that it may affect debugging experience (adding a frame), it's preferable to avoid function call, doesn't have any real drawback.

I had originally started using getFrame but then after looking at the cgen'd code it seemed better to do what I did; 
eg it avoids
```c
static N_INLINE(TFrame*, getFrame)(void) {
	TFrame* result;
	result = (TFrame*)0;
	result = framePtr__HRfVMH3jYeBJz6Q6X9b6Ptw;
	return result;
}

...
		TFrame* frX60gensym6851214_;
		frX60gensym6851214_ = getFrame();
```

vs
```c
		TFrame* fr;
		fr = (TFrame*)0; // btw. I think codegen should generally skip these assignments when there's a guaranteed assignment later
		fr = &FR_;
```

seems better/simpler even if it `getFrame` gets actually inlined; the emit uglyness is isolated in one place, so that seems fine


2. Shouldn't this be var framePtr {.importc: "FR_", nodecl.}: PFrame; framePtr?

this doesn't work, you can try for yourself starting from my PR and applying your suggestion inside `setFrameMsg` then running `nim c -r ./tests/stdlib/tstackframes.nim`; you'll get C errors; and if you do your change inside `getPFrame` and call getPFrame from setFrameMsg, it'll SIGSEGV

Note that I'm intentially exporting `getPFrame` but not using it in inside setFrameMsg as it results in cgen code bloat making sources less readable eg:
```
		{
			TFrame* framePtr;
			framePtr = (TFrame*)0;
			nimln_(16, "stackframes.nim");
			framePtr = &FR_;
			frX60gensym6835014_ = framePtr;
		}
```
even if C compiler will optimize away, it still adds overhead in some cases eg with `linedir`